crates/rpc-engine-types/src/optimism.rs

```diff
@@ -24,6 +24,29 @@ pub struct OptimismPayloadAttributes {
     pub gas_limit: Option<u64>,
 }
 
+/// This structure maps on the ExecutionPayloadV3 structure of the beacon chain spec.
+///
+/// See also: <https://github.com/ethereum/execution-apis/blob/6709c2a795b707202e93c4f2867fa0bf2640a84f/src/engine/shanghai.md#executionpayloadv2>
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
```

Comment:
this should also reference the op spec:

https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/exec-engine.md#engine_getpayloadv3

Reply:
> this should also reference the op spec:

sure, proposed here https://github.com/alloy-rs/alloy/pull/244